routing/router.go

```diff
@@ -922,6 +922,9 @@ type LightningPayment struct {
 	// the first hop.
 	PaymentHash [32]byte
 
+	// The array of routes which might be set by the sender to execute
```

Comment:
With this change, we can mostly leave `SendPayment` as it currently, with it unconditionally using the `SendToRoute` function with the routes discovered.